src/AutoRest.CSharp.V3/CodeGen/ClientWriter.cs

```diff
@@ -96,49 +105,81 @@ private void WriteParameter(CodeWriter writer, ServiceClientParameter clientPara
             writer.AppendRaw(",");
         }
 
+        private string CreateMethodName(ClientMethod operation, bool async, bool supportsPaging) => $"{operation.Name}{(supportsPaging ? "FirstPage" : string.Empty)}{(async ? "Async" : string.Empty)}";
 
-        private string CreateRequestMethodName(ClientMethod operation) => $"Create{operation.Name}Request";
+        private string CreateRequestMethodName(ClientMethod operation) => $"Create{operation.Name}{(operation.Paging != null ? "FirstPage" : string.Empty)}Request";
 
-        private void WriteRequestCreation(CodeWriter writer, ClientMethod operation)
-        {
+        private string CreateNextPageRequestMethodName(ClientMethod operation) => $"Create{operation.Name}NextPageRequest";
+
+        private string GetNextPageMethodName(ClientMethod operation) => $"{operation.Name}NextPageAsync";
+
+        private bool IsNextPageParameter(ServiceClientParameter parameter) =>
+            parameter.Location != ParameterLocation.Path
+            && parameter.Location != ParameterLocation.Body
+            && parameter.Location != ParameterLocation.Query;
 
-            writer.Append($"internal {typeof(HttpMessage)} {CreateRequestMethodName(operation)}(");
-            foreach (ServiceClientParameter clientParameter in operation.Parameters)
+        private void WriteRequestCreation(CodeWriter writer, ClientMethod operation, bool nextPage = false)
+        {
+            var methodName = nextPage ? CreateNextPageRequestMethodName(operation) : CreateRequestMethodName(operation);
+            writer.Append($"internal {typeof(HttpMessage)} {methodName}(");
+            var parameters = nextPage
+                ? operation.Parameters.Where(IsNextPageParameter).ToArray()
+                : operation.Parameters;
+            foreach (ServiceClientParameter clientParameter in parameters)
             {
                 writer.Append($"{_typeFactory.CreateInputType(clientParameter.Type)} {clientParameter.Name},");
             }
+            if (nextPage)
```

Comment:
I'm not sure I like this logic being on the writer side. Why not create another method definition on the  builder side and avoid changing the writer?

Reply:
I don't like it here either. But, for now, it works and passes the tests, which will allow me to refactor much easier. This PR is to get a baseline for paging working.